sstables/sstables.cc

```diff
@@ -1043,7 +1043,7 @@ future<> sstable::read_simple(T& component, const io_priority_class& pc) {
         auto fut = fi.size();
         return fut.then([this, &component, fi = std::move(fi)] (uint64_t size) {
             auto r = make_lw_shared<file_random_access_reader>(std::move(fi), size, sstable_buffer_size);
-            auto fut = parse(_version, *r, component);
+            auto fut = parse(_schema, _version, *r, component);
             return fut.finally([r] {
```

Comment:
For example, this caller is safe since it picks up the schema from the sstable, and an sstable's schema doesn't change.